### PR TITLE
RKMilCommute GPU compatibility

### PR DESCRIPTION
**Pull request by deeepeshthakur**
@ChrisRackauckas 
This looks good and is passing the tests
#167

### CODE REVIEW

File: src/perform_step/low_order.jl [hunk 1]
```diff
@@ -188,18 +188,17 @@ end
...
   ggprime_norm = 0.0
 
   @.. mil_correction = zero(u)
-  for i=1:length(dW),j=1:length(dW)
-      I[j,i] = 0.5*dW[i]*dW[j]
-      if alg_interpretation(integrator.alg) == :Ito
-        j == i && (I[i,i] -= 0.5*dt) # Ito correction
-      end
+  if alg_interpretation(integrator.alg) == :Ito
+    J = 0.5 .* (vec(dW) .* vec(dW)' - dt .* I)
```
Comment (ChrisRackauckas) [inline src/perform_step/low_order.jl]: This is allocating, and doesn't properly fuse because of the missing `.` on the `-`.

Comment (ChrisRackauckas) [inline src/perform_step/low_order.jl]:
```suggestion
    J .= 0.5 .* (vec(dW) .* vec(dW)' .- dt .* I)
```

File: src/perform_step/low_order.jl [hunk 2]
```diff
@@ -188,18 +188,17 @@ end
...
+  else
+    J = 0.5 .* vec(dW) .* vec(dW)'
```
Comment (ChrisRackauckas) [inline src/perform_step/low_order.jl]:
```suggestion
    J .= 0.5 .* vec(dW) .* vec(dW)'
```

File: src/caches/basic_method_caches.jl [hunk 3]
```diff
@@ -110,8 +110,9 @@ function alg_cache(alg::RKMilCommute,prob,u,ΔW,ΔZ,p,rate_prototype,noise_rate_
   du1 = zero(rate_prototype); du2 = zero(rate_prototype)
   K = zero(rate_prototype); gtmp = zero(noise_rate_prototype);
   L = zero(noise_rate_prototype); tmp = zero(rate_prototype)
-  I = zeros(length(ΔW),length(ΔW));
-  Dg = zeros(length(ΔW),length(ΔW)); mil_correction = zero(rate_prototype)
+  J = false .* vec(ΔW) .* vec(ΔW)'
```
Comment (ChrisRackauckas) [inline src/caches/basic_method_caches.jl]: If it's J then it will clash with the common name for the Jacobian. Make it WikJ?

Comment (deeepeshthakur) [reply src/caches/basic_method_caches.jl]: Okay!

File: src/perform_step/low_order.jl [hunk 4]
```diff
@@ -196,7 +196,7 @@ end
 
   @.. mil_correction = zero(u)
   if alg_interpretation(integrator.alg) == :Ito
-    WikJ .= 0.5 .* (vec(dW) .* vec(dW)' .- dt .* I)
+    WikJ .= 0.5 .* (vec(dW) .* vec(dW)'  .- dt .* Diagonal(fill(one(eltype(W.dW)),length(W.dW))))
```
Comment (ChrisRackauckas) [inline src/perform_step/low_order.jl]: That's going to allocate. Use Eye from FillArrays.jl

**Comment by ChrisRackauckas** [comment]
Going to merge, but the performance enhancement should come in a follow up PR

File: src/perform_step/low_order.jl (hunk 4 above)
Comment (deeepeshthakur) [reply src/perform_step/low_order.jl]: Will check this out!

**Pull request merged**